server/src/main/java/io/crate/planner/operators/CorrelatedJoin.java

```diff
@@ -163,18 +163,16 @@ public Void visitOuterColumn(OuterColumn outerColumn, Void ignored) {
         if (inputPlan == newInputPlan) {
             return this;
         }
-        List<Symbol> newOutputs = new ArrayList<>();
-        for (Symbol output: outputs) {
-            if (toCollect.contains(output)) {
-                newOutputs.add(output);
-            }
+        LinkedHashSet<Symbol> newOutputs = new LinkedHashSet<>();
+        for (Symbol outputToKeep : outputsToKeep) {
+            SymbolVisitors.intersection(outputToKeep, outputs, newOutputs::add);
```

Comment:
This change pretty much reminds me the recent change in the `Collect`. 

Can we have a problem, similar to https://github.com/crate/crate/pull/16125 if `CorrelatedJoin` has a `Rename` parent? Or it's supposed to be covered by the change in `SymbolVisitors`?

Reply:
If there is an issue with `Rename` as parent then we have it multiple times.

Reply:
See NestedLoopJoin, HashJoin, Filter. We have similar logic to https://github.com/crate/crate/pull/16125 in https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/planner/operators/Union.java#L207-L226, so maybe it is worth investigating to incorporate the logic of keeping the same object in `SymbolVisitors`.

Reply:
I think this is outside the scope of this pr.

Reply:
@mfussenegger WDYT ? Do we need a follow-up here ?

Reply:
I'm getting the impression that the `intersection` method should call the consumer with a symbol from `outputs`, so that the logic here as is just works